packages/teleport/src/components/MfaDeviceList/MfaDeviceList.tsx

```diff
@@ -0,0 +1,150 @@
+  devices = [],
+  onDelete,
+  isRecoveryFlow,
+}: Props) {
+  const [sortDir, setSortDir] = useState<Record<string, string>>({
+    registeredDate: SortTypes.ASC,
+  });
+
+  const mostRecentDevice = useMemo(() => sort(devices)[0], []);
+
+  function sort(data) {
+    const columnKey = Object.getOwnPropertyNames(sortDir)[0];
+    const sorted = sortBy(data, columnKey);
+    if (sortDir[columnKey] === SortTypes.ASC) {
+      return sorted.reverse();
+    }
+
+    return sorted;
+  }
+
+  function onSortChange(columnKey: string, sortDir: string) {
+    setSortDir({ [columnKey]: sortDir });
+  }
+
+  const data = sort(devices);
+
+  return (
+    <StyledTable
+      data={data}
+      style={isRecoveryFlow ? { overflow: 'hidden', borderRadius: '8px' } : {}}
+    >
+      <Column
+        columnKey="description"
+        cell={<TextCell />}
+        header={<Cell>Type</Cell>}
+      />
+      <Column
+        columnKey="name"
+        cell={<NameCell />}
+        header={<Cell>Device Name</Cell>}
+      />
+      <Column
+        columnKey="registeredDate"
+        header={
+          <SortHeaderCell
+            sortDir={sortDir.registeredDate}
+            onSortChange={onSortChange}
+            title="Registered"
+          />
+        }
+        cell={<DateCell />}
+      />
+      <Column
+        columnKey="lastUsedDate"
+        header={
+          <SortHeaderCell
+            sortDir={sortDir.lastUsedDate}
+            onSortChange={onSortChange}
+            title="Last Used"
+          />
+        }
+        cell={<DateCell />}
+      />
+      <Column
+        header={<Cell />}
+        cell={
+          <DeleteCell
+            onDelete={onDelete}
+            mostRecentDevice={mostRecentDevice}
+            isRecoveryFlow={isRecoveryFlow}
+          />
+        }
+      />
+    </StyledTable>
+  );
+}
+
+const NameCell = props => {
+  const { data, rowIndex } = props;
+  const { name } = data[rowIndex];
+
+  return (
+    <Cell title={name}>
+      <Text
+        style={{
+          maxWidth: '96px',
+          whiteSpace: 'nowrap',
+        }}
+      >
+        {name}
+      </Text>
+    </Cell>
+  );
+};
+
+const DateCell = props => {
+  const { data, rowIndex, columnKey } = props;
+  const dateText = displayDate(data[rowIndex][columnKey]);
+
+  return <Cell>{dateText}</Cell>;
+};
+
+const DeleteCell = props => {
+  const { data, rowIndex, onDelete, mostRecentDevice, isRecoveryFlow } = props;
+  const { id, name } = data[rowIndex];
+
+  if (id === mostRecentDevice.id && isRecoveryFlow) {
+    return null;
+  }
+
+  return (
+    <Cell align="right">
+      <ButtonBorder size="small" onClick={() => onDelete({ id, name })}>
+        Remove
+      </ButtonBorder>
+    </Cell>
+  );
+};
+
+type Props = {
+  devices: MfaDevice[];
+  onDelete({ id, name }: { id: string; name: string }): void;
+  // Whether this list is being shown in the recovery flow as opposed to in the account dashboard
+  isRecoveryFlow: boolean;
```

Comment:
is it really needed for this shared component to be aware of recovery flow and account dashboard?

Reply:
revised